stripe/http_client.py

```diff
@@ -89,10 +94,66 @@ def __init__(self, verify_ssl_certs=True, proxy=None):
                     " ""https"" and/or ""http"" keys.")
         self._proxy = proxy.copy() if proxy else None
 
+    def request_with_retries(self, method, url, headers, post_data=None):
+        num_retries = 0
+
+        while True:
+            try:
+                num_retries += 1
+                response = self.request(method, url, headers, post_data)
+                connection_error = None
+            except error.APIConnectionError as e:
+                connection_error = e
+                response = None
+
+            if self._should_retry(response, connection_error, num_retries):
+                self._sleep(num_retries)
+            else:
+                if response is not None:
+                    return response
+                else:
+                    raise connection_error
```

Comment:
Can we add a little logging here so that it's possible to get some visibility into what's happening for users that care about it?

Use stripe-go for inspiration, but I'd suggest at the least:

1. Emit an `error` line for the error that was received (although possibly only if we're going to retry if this would result in a duplicate line from elsewhere).
2. An `info` line something like this that announces its intention to sleep (from Go):

``` go
			s.Logger.Printf("Initiating retry %v for request %v %v%v after sleeping %v\n",
				retry, req.Method, req.URL.Host, req.URL.Path, sleepDuration)
```

If you check `util`, we have methods like `log_info`, `log_error` already built in.

Reply:
@brandur-stripe, `log_error` doesn't exist[0]-- are there particular reasons we don't implement so? Otherwise, I can add a method to it. 
If so should it look something like this; I'm concerned about the `_console_log_level()` line
```
def log_error(message, **params):
    msg = logfmt(dict(message=message, **params))
    if _console_log_level() in ['debug', 'info', 'error']: # added here?
        print(msg, file=sys.stderr)
    logger.error(msg) #[1] built-in python logger?
# existing method
def log_info(message, **params):
    msg = logfmt(dict(message=message, **params))
    if _console_log_level() in ['debug', 'info']:
        print(msg, file=sys.stderr)
    logger.info(msg)
```
[0] https://github.com/stripe/stripe-python/blob/master/stripe/util.py#L58
[1] https://docs.python.org/2/library/logging.html#logging.Logger.error

Reply:
Oh, weird! Yeah, we should probably add that, but I wouldn't worry about it for now (we'll do it in another PR) — just use `log_info`.